### PR TITLE
Fix Bug 1129624 - Refactor the function to get download links

### PR DESCRIPTION
**Pull request by kyoshino**
#2741 cannot be reopened so let me try again...

Assumption: Firefox for iOS and Thunderbird are coming to Bedrock sometime soon.

* Merge part of `mozorg/helpers/download_buttons.py` into `firefox/firefox_details.py` along with the corresponding tests
* Merge all Firefox-related helpers into `firefox/helpers.py` along with the corresponding tests
* Rename `FirefoxDetails` to `FirefoxDesktop`, `MobileDetails` to `FirefoxAndroid`
* Change the internal channel name `aurora` to product-neutral `alpha`
* Update the product-related URL functions accordingly

### CODE REVIEW

**Comment by kyoshino** [comment]
I'll combine the fix for [Bug 937865 - Simplify download buttons using the firefox-latest product in bouncer](https://bugzilla.mozilla.org/show_bug.cgi?id=937865)

**Comment by pmclanahan** [comment]
I see that the `mobile_details.json` file is all about Android, but I assume we'll need some data like this for iOS. I'm thinking naming the class `FirefoxMobile` would be better for now. We can always rename whenever we learn what exactly the iOS data looks like though. Hmm... 

I guess we'll leave it `FirefoxAndroid` for now since that's more accurate right now and revisit when we make the changes for iOS.

**Comment by pmclanahan** [comment]
I've skimmed the changes. It's a lot :) I'm deploying to demo1 now so we can take a look at how it works.

**Comment by kyoshino** [comment]
Thanks @pmclanahan!

**Comment by jgmize** [comment]
we have the mcom-tests running here: https://webqa-ci.mozilla.com/job/mozilla.demo1.saucelabs/1/console  -- already seeing a few failures, waiting for the job to finish for full output.

**Comment by kyoshino** [comment]
Looks like the product-details lib will be gone in favour of the new Ship-It system.

https://bugzilla.mozilla.org/show_bug.cgi?id=939470#c38
https://bugzilla.mozilla.org/show_bug.cgi?id=1083718
https://wiki.mozilla.org/ReleaseEngineering/Applications/Ship_It

**Comment by pmclanahan** [comment]
@kyoshino we hope it will. No idea what a realistic timeline for that is though. For now I think we should assume that we'll need to use the product-details JSON files like we are now for the foreseeable future.

**Comment by pmclanahan** [comment]
I think we're unblocked here. Testing seems to have gone well according to comments in the bug. We ready to merge? @kyoshino @jgmize ?

**Comment by jgmize** [comment]
The mcom-test run that failed that I mentioned in https://github.com/mozilla/bedrock/pull/2815#issuecomment-78145403 got deleted, and something has completely broken the old jenkins jobs so I'm having trouble getting them to run there, but I'd like to get a green run from mcom-tests before we merge this-- I should hopefully have an answer on that tomorrow.

**Comment by cargoworker** [comment]
Can one of the admins verify this patch?

**Comment by jgmize** [comment]
Here are the results of running the current master branch of mcom-tests against demo1: https://ci.masterfirefoxos.com/view/www.mozilla.org/job/www-demo1.allizom.org-saucelabs/2/console

**Comment by kyoshino** [comment]
Thank you @jgmize, I'll take a look later today or tomorrow.

**Comment by kyoshino** [comment]
I sent https://github.com/mozilla/mcom-tests/pull/394 to fix the first 2 failures. Other 2 issues were in my code, where filtering the builds on the /all/ pages wasn't working. Fixed that issue in my follow-up commit.

**Comment by jgmize** [comment]
Thanks for all your hard work and patience on this @kyoshino. The code is good, it just needed a rebase, which I went ahead and did in #2871.
